system/HTTP/URI.php

```diff
@@ -702,7 +702,7 @@ public function setQuery(string $query)
 				continue;
 			}
 
-			$parts[$key] = $value;
+			$parts[$key] = urldecode($value);
```

Comment:
If it be accepted, maybe also is needed to decode the keys. Looking for this https://github.com/bcit-ci/CodeIgniter4/commit/2604ea75b7938e0cfa5efdfb022c8ab14608965a .

Since the user waits (normally) by a non-encoded string as key:

By it: `$this->request->getGet('a b')`

And not by it `$this->request->getGet('a%2Bb')`

Reply:
```php
$uri = new \CodeIgniter\HTTP\URI('http://www.example.com/some/path?a+b= a+ asss55&asad= a7dsa');
var_dump((string)$uri);
var_dump($uri);
exit;
```

```
/application/Controllers/Blog/Search.php:50:string 'http://www.example.com/some/path?a%2Bb=+a++asss55&asad=+a7dsa' (length=61)

/application/Controllers/Blog/Search.php:51:
object(CodeIgniter\HTTP\URI)[155]
  protected 'uriString' => null
  protected 'segments' => 
    array (size=2)
      0 => string 'some' (length=4)
      1 => string 'path' (length=4)
  protected 'scheme' => string 'http' (length=4)
  protected 'user' => null
  protected 'password' => null
  protected 'host' => string 'www.example.com' (length=15)
  protected 'port' => null
  protected 'path' => string '/some/path' (length=10)
  protected 'fragment' => string '' (length=0)
  protected 'query' => 
    array (size=2)
      'a+b' => string ' a  asss55' (length=10)
      'asad' => string ' a7dsa' (length=6)
  protected 'defaultPorts' => 
    array (size=4)
      'http' => int 80
      'https' => int 443
      'ftp' => int 21
      'sftp' => int 22
  protected 'showPassword' => boolean false

```